wot.py

```diff
@@ -118,6 +118,25 @@ def start_round() -> None:
                 connection.poll(timeout_seconds=0.1)
             start_round()
 
+def truncate_username(username, max_length=12):
+    if len(username) <= max_length:
+        return username
+    
+    last_char = username[-1]
+    start_chars = 1
+    
+    while True:
+        omitted_count = len(username) - start_chars - 1
+        test_display = f"{username[:start_chars]}{omitted_count}{last_char}"
+        
+        if len(test_display) > max_length:
+            start_chars -= 1
+            omitted_count = len(username) - start_chars - 1
+            return f"{username[:start_chars]}{omitted_count}{last_char}"
+        elif omitted_count <= 0:
+            return username
+        else:
+            start_chars += 1
```

Comment:
The loop seems a bit silly

```suggestion
def truncate_username(username, max_length=12):
    if len(username) <= max_length:
        return username
    else:
        omitted_count = 1 + len(username) - max_length
        return f"{username[:max_length - 2]}{omitted_count}{username[-1]}"
```